### PR TITLE
feat(mdChips): Add `md-on-select` expression support.

### LINKED ISSUE #3413: md-chips: add "md-selected-item-change" event/method and allow selection on readonly
I'm really missing a convenient way of knowing when a `md-chip` was selected, like `md-autocomplete` has implemented it.

```HTML
<md-chips ng-model="vm.chips"
          md-selected-item-change="vm.chipSelected(item)">
</md-chips>
```

Also you should be able to select chips when the control is set to readonly, the only situation where you shouldn't be able to do this in my opinion is when the control is disabled.

**EDIT:** same goes for static chips, also some way of setting the selected chip would be a nice to have.

### PR DESCRIPTION
**Pull request by topherfangio**
Users want a way to be notified when a chip is selected.

closes #3413

### CODE REVIEW

**Comment by topherfangio** [comment]
@typotter ready for review :-)

**Comment by typotter** [comment]
LGTM

File: src/components/chips/js/chipsController.js [hunk 1]
```diff
@@ -72,6 +72,13 @@ function MdChipsCtrl ($scope, $mdConstant, $log, $element, $timeout) {
    * @type {boolean}
    */
   this.useMdOnAppend = false;
+
+  /**
+   * Whether to use the mdOnSelect expression to notify the component's user
+   * after selecting a chip from the list.
+   * @type {boolean}
+   */
+  this.useMdOnSelect = false;
```
Comment (ThomasBurleson) [inline src/components/chips/js/chipsController.js]: This is not a good convention
*   `useMdOnAppend` => `useOnAppend`
*   `useMdOnRemove` => `useOnRemove`
*   `useMdOnSelectExpression` => `useOnSelectExpression`

File: src/components/chips/js/chipsDirective.js [hunk 2]
```diff
@@ -164,6 +165,7 @@
         secondaryPlaceholder: '@',
         mdOnAppend: '&',
         mdOnRemove: '&',
+        mdOnSelect: '&',
```
Comment (ThomasBurleson) [inline src/components/chips/js/chipsDirective.js]: Our convention: `onSelect :  '&mdOnSelect'`

**Comment by topherfangio** [comment]
@ThomasBurleson Updated based on feedback, should be ready for merge :-)

**Pull request closed without merging**